src/audio/pipeline.c

```diff
@@ -720,7 +727,7 @@ static int pipeline_comp_copy(struct comp_dev *current, void *data, int dir)
 	}
 
 	/* copy to downstream immediately */
-	if (dir == PPL_DIR_DOWNSTREAM) {
+	if (dir == PPL_DIR_DOWNSTREAM && cpu_is_me(current->comp.core)) {
```

Comment:
I feel this change can be separated, only running "copy" on the components on the current core but walking the entire pipeline?

Reply:
Extracted to separate commit.